pkg/graph/builder.go

```diff
@@ -397,10 +403,8 @@ func (b *Builder) buildDependencyGraph(
 				resource.addDependencies(resourceDependencies...)
 				resourceVariable.AddDependencies(resourceDependencies...)
 				// We need to add the dependencies to the graph.
-				for _, dependency := range resourceDependencies {
-					if err := directedAcyclicGraph.AddEdge(resourceName, dependency); err != nil {
-						return nil, err
-					}
+				if err := directedAcyclicGraph.AddDependencies(resource.id, resourceDependencies); err != nil {
```

Comment:
This should be (very) marginally faster than adding one at a time, but really it was because I was confused on the direction of the "dependency arrow"